test/conformance/enqueue/urEnqueueUSMMemcpy2D.cpp

```diff
@@ -1,2 +1,167 @@
+            urUSMDeviceAlloc(context, device, nullptr, num_elements, 0, &pDst));
+        ur_event_handle_t memset_event = nullptr;
+        ASSERT_SUCCESS(urEnqueueUSMMemset2D(queue, pSrc, pitch, memset_value,
+                                            width, height, 0, nullptr,
+                                            &memset_event));
+        ASSERT_SUCCESS(urQueueFlush(queue));
+        ASSERT_SUCCESS(urEventWait(1, &memset_event));
+        ASSERT_SUCCESS(urEventRelease(memset_event));
+    }
+
+    void TearDown() override {
+        uur::urQueueTestWithParam<uur::TestParameters2D>::TearDown();
+        if (pSrc) {
+            ASSERT_SUCCESS(urUSMFree(context, pSrc));
+        }
+        if (pDst) {
+            ASSERT_SUCCESS(urUSMFree(context, pDst));
+        }
+    }
+
+    void verifyMemcpySucceeded() {
+        std::vector<uint8_t> host_mem(pitch * height);
+        ASSERT_SUCCESS(urEnqueueUSMMemcpy2D(queue, true, host_mem.data(), pitch,
+                                            pDst, pitch, width, height, 0,
+                                            nullptr, nullptr));
+        for (size_t w = 0; w < width; ++w) {
+            for (size_t h = 0; h < height; ++h) {
+                const auto *host_ptr = host_mem.data();
+                const size_t index = (pitch * h) + w;
+                ASSERT_TRUE(*(host_ptr + index) == memset_value);
+            }
+        }
+    }
+
+    void *pSrc = nullptr;
+    void *pDst = nullptr;
+    static constexpr int memset_value = 42;
+    size_t pitch = 0;
+    size_t width = 0;
+    size_t height = 0;
+};
+UUR_TEST_SUITE_P(urEnqueueUSMMemcpy2DTestWithParam,
+                 ::testing::Values(uur::TestParameters2D{1, 1, 1}),
+                 uur::print2DTestString<urEnqueueUSMMemcpy2DTestWithParam>);
+
+TEST_P(urEnqueueUSMMemcpy2DTestWithParam, SuccessBlocking) {
+    ASSERT_SUCCESS(urEnqueueUSMMemcpy2D(queue, true, pDst, pitch, pSrc, pitch,
+                                        width, height, 0, nullptr, nullptr));
+    ASSERT_NO_FATAL_FAILURE(verifyMemcpySucceeded());
+}
+
+TEST_P(urEnqueueUSMMemcpy2DTestWithParam, SuccessNonBlocking) {
+    ur_event_handle_t memcpy_event = nullptr;
+    ASSERT_SUCCESS(urEnqueueUSMMemcpy2D(queue, false, pDst, pitch, pSrc, pitch,
+                                        width, height, 0, nullptr,
+                                        &memcpy_event));
+    ASSERT_SUCCESS(urQueueFlush(queue));
+    ASSERT_SUCCESS(urEventWait(1, &memcpy_event));
+    const auto eventStatus = uur::GetEventInfo<ur_event_status_t>(
+        memcpy_event, UR_EVENT_INFO_COMMAND_EXECUTION_STATUS);
+    ASSERT_TRUE(eventStatus.has_value());
+    ASSERT_EQ(eventStatus.value(), UR_EVENT_STATUS_COMPLETE);
+
+    ASSERT_NO_FATAL_FAILURE(verifyMemcpySucceeded());
+}
+
+using urEnqueueUSMMemcpy2DNegativeTest = urEnqueueUSMMemcpy2DTestWithParam;
+UUR_TEST_SUITE_P(urEnqueueUSMMemcpy2DNegativeTest,
+                 ::testing::Values(uur::TestParameters2D{16, 16, 16}),
```

Comment:
It's going to make naming this suite tricky (`urEnqueueUSMMemcpy2D16x16Test`?) but would it be worth putting in a test that checks for success with these dimensions?

Reply:
Good spot. I actually forgot to add the parametrized versions for the success case. I've changed the negative tests to use {1,1,1,} which is a tested success case.